artemis-protocols/artemis-mqtt-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/mqtt/MQTTUtil.java

```diff
@@ -67,6 +76,8 @@
 
    public static final int DEFAULT_KEEP_ALIVE_FREQUENCY = 5000;
 
+   public static final String logUTF8PublishPayload = System.getProperty("org.apache.activemq.artemis.core.protocol.mqtt.logUTF8PublishPayload");
```

Comment:
this is about tracing.. why don't you just always log the UTF?
if you don't need that level of information, just take trace out.


of you could do something like.. have this whole thing at debug... and if trace, throw the body on log.trace.

The system property is a bit odd here.

Reply:
My 2 cents. The PUBLISH payload has a maximum size of 256 MB. I could have a use case where I need tracing packets but not printing 256 MB of bytes on the console. Use case where could be a problem in the messages flow that is not dependent on the payload. In this moment using just TRACE enabled or not it's just about having or not the logging. Maybe we could use DEBUG for normal packets logging and then TRACE for printing the payload as well.

Reply:
To keep this as simple as possible I'll get rid of the system property and always log the payload as UTF-8, but also always limit the output to a certain number of characters so we don't get hundreds of megs of data (potentially).

Reply:
We do that in 5.x for things like logs on STOMP messages where you don't want to dump the full content portion of the frame but a small bit can help, I can't recall what the default is but something like 60 chars or the like is used.  You end up with a content string of "ABCD....ZYZ"

Reply:
So for people who need to have the entire payload because it's in a reasonable range size, they need to use Wireshark ?